plugins/nf-amazon/src/main/com/upplication/s3fs/S3OutputStream.java

```diff
@@ -253,6 +255,9 @@ private MessageDigest createMd5() {
      */
     @Override
     public void write (int b) throws IOException {
+        if( closed ){
+            throw new IOException("Can't write into a closed stream");
```

Comment:
Do you mean, adding this it makes the download fail and retry?

Reply:
no, it aborts the writing if some part, after retries, fails

I mean:
- stream is sending bytes
- we are sending in background bunch of bytes
- some of them fail (networks down for example), and we retry this part several times and mark as closed if all retries fail 
- meanwhile, stream was sending more bytes but as soon we detected some parts failed we abort and avoid to send remainder streaming because it'll fail due at least 1 part is missing

Reply:
Go, it makes fail fast as in the title of the PR :)